src/projections/genericsom.cpp

```diff
@@ -221,8 +216,62 @@ PJ *PROJECTION(misrsom) {
     Q->c1 /= 15.;
     Q->c3 /= 45.;
 
-    P->inv = misrsom_e_inverse;
-    P->fwd = misrsom_e_forward;
+    P->inv = genericsom_e_inverse;
+    P->fwd = genericsom_e_forward;
 
    return P;
 }
+
+PJ *PROJECTION(genericsom) {
+    struct pj_opaque *Q = static_cast<struct pj_opaque*>(calloc (1, sizeof (struct pj_opaque)));
+    if (nullptr==Q)
+        return pj_default_destructor (P, PROJ_ERR_OTHER /*ENOMEM*/);
+    P->opaque = Q;
+
+    // ascending longitude (radians)
+    P->lam0 = pj_param(P->ctx, P->params, "dasc_lon").f;
+    if (P->lam0 < -M_TWOPI || P->lam0 > M_TWOPI)
```

Comment:
it is quite untypical for PROJ angular projection parameters to be specified in radians. They are rather specified in degrees. If you use "rasc_lon" it will expect degrees as the input but will return the result converted in radians
Same remark for inclination angle

Reply:
It is stored internally and displayed to users as radians in the data products, so that's the direction I went in since it saves users a conversion. If it isn't expected behavior here though, I can change it (and the tests) to match.

Reply:
Maybe @kbevers want to chime in ?

Reply:
As some additional context, I'll mention that GCTP takes the values in radians which might be why this was adopted for the MISR output products as well (i.e. so that the projection parameters specified in the products could be directly used to init GCTP projections).
The [MISR DPS](https://asdc.larc.nasa.gov/documents/misr/DPS_v50_RevS.pdf) mentions:
"[GCTP's] for_init takes the projection parameters from step a. as arguments" where "step a" refers to the result of [HDF-EOS GDprojinf()](https://hdfeos.org/examples/c_grid_lonlat.php) used on the MISR data products.

That being said, I'm still happy to change it to degrees and document the same if that's the preferred convention to use here.

Reply:
I think it would be confusing to use radians as input here. At least as the only option. You could possibly implement options for both degrees (`+asc_lon`) and radians (`+asc_lon_rad`). They would of course have to be mutally exclusive.

Reply:
> You could possibly implement options for both degrees (`+asc_lon`) and radians (`+asc_lon_rad`).

Actually looking at the dmstor_ctx() function (which is used by pj_param("r{parameter_name}")), I've discovered a hidden feature (at least to me, but I see from commit history that it dates back to the initial commit of PROJ4!). If the value is immediately followed by 'r' or 'R' , it mean radians (you may put anything after the r

```
$ echo 0 0 | bin/cs2cs +proj=longlat +to +proj=longlat +lon_0=1.5707963267948966r
90dW	0dN 0.000

$ echo 0 0 | bin/cs2cs +proj=longlat +to +proj=longlat +lon_0=1.5707963267948966radwhatever
90dW	0dN 0.000

$ echo 0 0 | bin/cs2cs +proj=longlat +to +proj=longlat +lon_0=90
90dW	0dN 0.000
```

So there's nothing to do but grabbing the parameter with pj_param(P->ctx, P->params, "rasc_lon").f to trigger the use of dmstor_ctx()

Reply:
Well isn't that just lovely! Solves the issue at hand it would seem. Just make sure to mention it in the docs for the projection so that users are made aware of that option.

Related question: Where do we document this generic feature? Somewhere in https://github.com/OSGeo/PROJ/blob/master/docs/source/usage/projections.rst perhaps?

Reply:
> Somewhere in https://github.com/OSGeo/PROJ/blob/master/docs/source/usage/projections.rst perhaps?

would be a good idea.  Formats like ``1d45'60''W`` are not documented too

Reply:
Thanks for the feedback! I have updated the radian input arguments to the "r" (degrees) parameter type and updated the docs to match as well as adding tests for both decimal degrees and radian inputs (using the "r" suffix to the input value).

Please let me know if this is what you were thinking or if you'd like anything further.